modules/objdetect/src/aruco/aruco_board.cpp

```diff
@@ -468,7 +487,11 @@ CharucoBoard::CharucoBoard(const Size& size, float squareLength, float markerLen
     for(int y = 0; y < size.height; y++) {
         for(int x = 0; x < size.width; x++) {
 
-            if(y % 2 == x % 2) continue; // black corner, no marker here
+            if(legacy) {
```

Comment:
actually, this is the only important bit. Generating board images in the legacy format is nice for consistency with the flag, but not really desirable.

Could we introduce a function that shuffles `impl->objPoints` from new to old layout? This would fix detection of existing boards which is the main issue here.

Reply:
```impl->objPoints``` is also used further down in the constructor in ```calcNearestMarkerCorners()```. So reshuffling also needs to include to clear whatever is going on there and execute it again. This is what I meant before with clearing and recreating the board again. Sure it's possible to do, but not really elegant imho. But I will check and make a proposal.